libconnman-qt/networkmanager.cpp

```diff
@@ -530,12 +530,7 @@ NetworkService* NetworkManager::defaultRoute() const
 
 NetworkTechnology* NetworkManager::getTechnology(const QString &type) const
 {
-    if (m_technologiesCache.contains(type))
-        return m_technologiesCache.value(type);
-    else {
-        qDebug() << "Technology " << type << " doesn't exist";
-        return NULL;
-    }
+    return m_technologiesCache.value(type);
```

Comment:
This m_technologiesCache table contains pointers, not NetworkTechnology objects. By default value() will return NULL which is fine. I'm not quite sure how default-constructed pointer values in QHashTable become NULL but they do.